sound/soc/sof/topology.c

```diff
@@ -1428,6 +1445,43 @@ static int sof_connect_dai_widget(struct snd_soc_component *scomp,
 	return 0;
 }
 
+static struct sof_ipc_comp *sof_comp_setup(struct snd_sof_widget *swidget,
+					   size_t *ipc_size, int index,
+					   int core)
+{
+	struct sof_ipc_comp_new_ext nul_ext_data = {0};
+	struct sof_ipc_comp *comp;
+	size_t total_size = *ipc_size;
+
+	/* check if extended data valid */
+	if (!memcmp(&swidget->comp_ext, &nul_ext_data,
```

Comment:
I think you mean: "check whether extended data is _present_"? all-zeroes doesn't seem like an invalid bug.

Even better, explicitly point to the other code that this critically relies on:
```C
/* Check whether sof_X_Y_Z() put some extended data or the special, all-zeroes value */
if (!memcmp(&swidget->comp_ext, &nul_ext_data,
```

Reply:
Yes, can you help change this comment together in your update?

Reply:
Yes, it is here: https://github.com/thesofproject/linux/blob/b80c2f7d1d34dbcc03a74ac5681d72298876636b/sound/soc/sof/topology.c#L2343

Reply:
Yes, it is here:
https://github.com/thesofproject/linux/blob/b80c2f7d1d34dbcc03a74ac5681d72298876636b/sound/soc/sof/topology.c#L2343

Reply:
@keyonjie this check here is too confusing for me. The check makes sure the swidget->comp_ext is all zeroes before allocating memory for "comp". Why would you allocate memory for ext_data only if it is all zeroes? the UUID is not going to zero isnt it?

Reply:
> @keyonjie this check here is too confusing for me. The check makes sure the swidget->comp_ext is all zeroes before allocating memory for "comp". Why would you allocate memory for ext_data only if it is all zeroes? the UUID is not going to zero isnt it?

Thanks for spotting this, the logic here was wrong, should be opposite, correcting it now.